hedera-node/hedera-app/src/main/java/com/hedera/node/app/meta/MonoHandleContext.java

```diff
@@ -39,17 +40,20 @@ public class MonoHandleContext implements HandleContext {
     private final ExpiryValidator expiryValidator;
     private final TransactionContext txnCtx;
     private final AttributeValidator attributeValidator;
+    private final ReadableStoreFactory readableStoreFactory;
 
     @Inject
     public MonoHandleContext(
             @NonNull final EntityIdSource ids,
             @NonNull final ExpiryValidator expiryValidator,
             @NonNull final AttributeValidator attributeValidator,
-            @NonNull final TransactionContext txnCtx) {
+            @NonNull final TransactionContext txnCtx,
+            @NonNull final ReadableStoreFactory storeFactory) {
```

Comment:
Are you sure, this is correct? `MonoHandleContext` has singleton-scope while a `ReadableStoreFactory` is bound to a particular state.

In the upcoming design, which will be similar to what we have in pre-handle already, a context will be created per transaction and it makes sense to add `ReadableStoreFactory`. But here, I think it is wrong.

Reply:
Yeah, I have added as temporary solution, until `HandleContext` is added to be used in `HandleWorkflow`.
Do you have any other ideas how can I inject it for `createReadableStore` ?

Should I inject `ReadableStoreFactory`  in `UtilPrng.handle()` method and provide it to `createReadableStore` ? @netopyr